### PR TITLE
Added Dijkstra Algorithm

### LINKED ISSUE #53: Add Dijkstra algorithms
The python package currently doesn't cover Dijkstra Algorithm

- Dijkstra Algorithm - @Siddhant-K-code

### PR DESCRIPTION
**Pull request by Siddhant-K-code**

### CODE REVIEW

File: edualgo/graph_algorithms/Dijkstra_algorithms.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
+from .__init__ import print_msg_box
+
+import sys
+
+# Providing the graph
+vertices = [[0, 0, 1, 1, 0, 0, 0],
```
Comment (Abhijit2505) [inline edualgo/graph_algorithms/Dijkstra_algorithms.py]: We don't need hard coded input for our package, just pass it as a parameter to the function,

File: edualgo/graph_algorithms/Dijkstra_algorithms.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+i = 0
+
+# Printing the distance
+for distance in visited_and_distance:
+    print("Distance of ", chr(ord('a') + i),
+          " from source vertex: ", distance[1])
+    i = i + 1
+
+
+def print_Dijkstras(self):
```
Comment (Abhijit2505) [inline edualgo/graph_algorithms/Dijkstra_algorithms.py]: add a hint substring to this function name

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]
Please Make the Following Changes.

File: edualgo/graph_algorithms/Dijkstra_algorithms.py (hunk 1 above)
Comment (Siddhant-K-code) [reply edualgo/graph_algorithms/Dijkstra_algorithms.py]: Done

**Comment by Siddhant-K-code** [comment]
Done @Abhijit2505

**Comment by Abhijit2505** [comment]
> Done @Abhijit2505

So which module should the user call and pass the parameter to get the task done?

There are no functions, please have a look at the codes in other folders and try to compare what you have missed in this code. You have missed a crucial thing.

Hint - The function call

**Comment by Siddhant-K-code** [comment]
> 
> 
> > Done @Abhijit2505
> 
> So which module should the user call and pass the parameter to get the task done?
> 
> There are no functions, please have a look at the codes in other folders and try to compare what you have missed in this code. You have missed a crucial thing.
> 
> Hint - The function call

`import time`  or main() func. ??

**Comment by Abhijit2505** [comment]
> > > Done @Abhijit2505
> > 
> > 
> > So which module should the user call and pass the parameter to get the task done?
> > There are no functions, please have a look at the codes in other folders and try to compare what you have missed in this code. You have missed a crucial thing.
> > Hint - The function call
> 
> `import time` or main() func. ??

You have written the program in a competitive programming manner, but it's a software.

What I mean to say, all your code should be inside a function which a user can call from his/her program.

For a better understanding watch the demo videos.

Your entire code should be inside something like ``dijkstra_algorithm(graph,hint=False)``

**Comment by Abhijit2505** [comment]
> > > > Done @Abhijit2505
> > > 
> > > 
> > > So which module should the user call and pass the parameter to get the task done?
> > > There are no functions, please have a look at the codes in other folders and try to compare what you have missed in this code. You have missed a crucial thing.
> > > Hint - The function call
> > 
> > 
> > `import time` or main() func. ??
> 
> You have written the program in a competitive programming manner, but it's a software.
> 
> What I mean to say, all your code should be inside a function which a user can call from his/her program.
> 
> For a better understanding watch the demo videos.
> 
> Your entire code should be inside something like `dijkstra_algorithm(graph,hint=False)`

Also there is no need to take input, the user will pass it as a parameter, i.e. the graph and the edges. You just need to perform the main algorithm part and return the output to the user as soon as he/she calls it.

**Comment by Siddhant-K-code** [comment]
I think it's good now

File: edualgo/graph_algorithms/Dijkstra_algorithms.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
+from .__init__ import print_msg_box
+
+import sys
+
+
+def dijkstra_algorithm(graph, hint=False):
+
+    # Providing the graph
+    size = int(input())  # Size of the Vertices & Edges
```
Comment (Abhijit2505) [inline edualgo/graph_algorithms/Dijkstra_algorithms.py]: When the user has passed graph as parameter already why to take input explicitly for size ?

File: edualgo/graph_algorithms/Dijkstra_algorithms.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+    edges = []
+    for x in range(size):
+        edges.append([int(y) for y in input().split()])
+
+    # Find which vertex is to be visited next
+
+    global visited_and_distance
+
+    def to_be_visited(graph, hint=False):
```
Comment (Abhijit2505) [inline edualgo/graph_algorithms/Dijkstra_algorithms.py]: There is no need to assign functions inside functions, you can just take this function out of the `def dijkstra_algorithm(graph, hint=False):` and call it by passing the required parameters.

Remember ours is a learning library and we keep things clean and simple as much as possible.

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]

**Comment by Abhijit2505** [comment]
@Siddhant-K-code , No need to hurry, take your time, understand the codes in other folders, understand how our flow works and only after that, you should code the algorithm out.

Coding an algorithm is very simple, but fitting it into the software needs some time for a programmer. Take some time.

File: edualgo/graph_algorithms/Dijkstra_algorithms.py (hunk 3 above)
Comment (Siddhant-K-code) [reply edualgo/graph_algorithms/Dijkstra_algorithms.py]: then how we will take the input for vertices and edges ?

**Comment by Siddhant-K-code** [comment]
Check This Now !

**Review by Abhijit2505 — APPROVED** [review]

**Comment by Abhijit2505** [comment]
@all-contributors add @Siddhant-K-code for code

**Pull request merged**